File: FingerRace/SquareTarget.m
//
//  SquareTarget.m
//  FingerRace
//
//  Created by Emmett Butler on 3/31/13.
//
//

#import "SquareTarget.h"
#import "Player.h"

@implementation SquareTarget

-(id) initWithTexture:(CCTexture2D*)texture rect:(CGRect)rect{
    if( (self=[super initWithTexture:texture rect:rect])){
        
    }
    return self;
}

+ (id)spriteWithPlayer:(Player *)player{
    SquareTarget *sprite = [SquareTarget spriteWithFile:@"square.png"];
    [sprite setColor:player.color];
    [sprite setScale:4];
    return sprite;
}

-(void)wasTouched{
    [self runAction:
     [CCSequence actions:
      [CCScaleTo actionWithDuration:.13 scale:1.3],
      [CCScaleTo actionWithDuration:.13 scale:1],
      nil]
     ];
}

@end

File: FingerRace/Classes/SquareTarget.cpp
//
//  SquareTarget.cpp
//  CocosTest
//
//  Created by Evan Moore on 4/2/13.
//
//

#include "SquareTarget.h"
#include "Player.h"

#import "cocos2d.h"

SquareTarget::SquareTarget() {
    
}

void SquareTarget::initWithPlayer(Player *p) {
    this->initWithFile("square.png");
    this->setColor(p->color);
}

void SquareTarget::wasTouched() {
    cocos2d::CCScaleTo *scaleOne = new cocos2d::CCScaleTo();
    scaleOne->initWithDuration(0.13, 1.3);
    
    cocos2d::CCScaleTo *scaleTwo = new cocos2d::CCScaleTo();
    scaleTwo->initWithDuration(0.13, 1.0);
    
    cocos2d::CCSequence *sequence = new cocos2d::CCSequence();
    sequence->initWithTwoActions(scaleOne, scaleTwo);
}
